### PR TITLE
Support TP with MoE Expert Streaming

### PR DESCRIPTION
**Pull request by zyma98**
Allow `stream_routed_experts` with `tensor_parallel_size > 1` for GPT-OSS, Mixtral, Qwen3/3.5-MoE, and DeepSeek-V4 by emitting per-rank offline expert packs sized to `I_local`, so the streamer can page contiguous TP-local sections instead of strided HF extents. Storage program version changed to 6, with new pack kinds for GPT-OSS RoutedDecode TP MXFP4, Mixtral BF16, Qwen named/fused BF16, and DSv4 MXFP4. Also fix DeepSeek-V4 MQA KV sizing under TP so the memory planner no longer divides the single KV head by `tp_size` and computes zero page bytes.

## Summary by CodeRabbit

* **New Features**
  * Added tensor-parallel routed-expert streaming for GPT-OSS, DeepSeek-V4, Mixtral, Qwen3, and Qwen3.5 MoE models.
  * Added BF16 and MXFP4 expert-pack support with per-rank packing.
  * Improved model-specific tensor-parallel memory and KV-cache handling.
  * Added clearer validation for unsupported architecture and parallelism combinations.

* **Bug Fixes**
  * Improved CUDA stream usage, RoPE selection, synchronization options, and logits-capacity validation.
  * Updated streaming and storage compatibility versions.

### CODE REVIEW

File: driver/cuda/src/model/gpt_oss_routed_mxfp4_pack.cpp [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        }
+        ctx.local_intermediate = ctx.full_intermediate / tp;
+        ctx.local_start = rank * ctx.local_intermediate;
+        if (ctx.local_start % 32 != 0 || ctx.local_intermediate % 32 != 0) {
+            throw std::runtime_error(
+                "expert pack: RoutedDecode TP shard must align to 32");
+        }
+        ctx.hidden = ctx.gu_groups * 32;
+        ctx.down_groups_full = static_cast<int>(dn_w_info.shape[2]);
+        ctx.down_groups_local = ctx.local_intermediate / 32;
```
[the remark this answers is not in the document]
Comment (zyma98) [reply driver/cuda/src/model/gpt_oss_routed_mxfp4_pack.cpp:97]: Addressed with the new commit.

File: driver/weight_loader/src/stream_arch.rs [hunk 2]
```diff
@@ -212,10 +382,25 @@ fn align_up_u64(v: u64, a: u64) -> u64 {
...
+    let rank = i64::from(target.tp_rank);
+    if full % world != 0 {
+        return Err(CompileError::InvalidInput(format!(
+            "dimension {full} is not divisible by tp_size {}",
+            target.tp_size
+        )));
+    }
+    let local = full / world;
+    Ok((rank * local, local))
+}
```
[the remark this answers is not in the document]
Comment (zyma98) [reply driver/weight_loader/src/stream_arch.rs]: Addressed with the new commit.

File: driver/weight_loader/src/stream_arch.rs [hunk 3]
```diff
@@ -357,26 +673,29 @@ fn gpt_oss_eager_bf16_section_bytes(
...
              (gate_up {:?}, down {:?})",
             gate_up_blocks.shape, down_blocks.shape
         )));
     }
-    // BF16: rows * cols * 2.
+    // BF16: rows * cols * 2 (TP-local intermediate).
     let gate = intermediate * hidden * 2;
     let down = hidden * intermediate * 2;
     Ok([gate, gate, down])
 }
```
[the remark this answers is not in the document]
Comment (zyma98) [reply driver/weight_loader/src/stream_arch.rs:687]: Addressed with the new commit.
